simulator/src/integrated/mod.rs

```diff
@@ -50,17 +57,30 @@ pub(crate) async fn run_configured_simulation(
             producer,
             kafka_producer_thread_set: &mut kafka_producer_thread_set,
             topics: Topics {
-                traces: cli.trace_topic.as_deref().unwrap(),
-                events: cli.event_topic.as_deref().unwrap(),
-                frame_events: cli.frame_event_topic.as_deref().unwrap(),
-                run_controls: cli.control_topic.as_deref().unwrap(),
+                traces: cli
```

Comment:
This error checking would be better handled via `clap`.

Reply:
Running the simulator in different modes (i.e. cli subcommands) makes use of different combinations of these topics. For instance, they aren't all required for sending Run Commands, or emitting a simulated Trace using the original `Single` subcommand.

Reply:
You can still handle this with `clap`, [this](https://github.com/DanNixon/emfcamp-schedule-api/blob/main/cli/src/commands/upcoming.rs#L10) is an example of solving pretty much the same problem. i.e. you have several commands that use different combinations of the same arguments.